src/engagement_db_to_analysis/rapid_pro_advert_functions.py

```diff
@@ -190,17 +164,44 @@ def _get_uuids_to_sync(target_uuids, synced_uuids):
 
     return uuids_to_sync
 
+def _ensure_contact_field_exists(workspace_contact_fields, target_contact_field_label, rapid_pro):
+    """
+    Checks if a workspace contains the target contact field, creates one otherwise.
+
+    :param workspace_contact_fields: All contact fields in the rapid_pro workspace.
+    :type workspace_contact_fields: list of temba_client.v2.types.Field
+    :param target_contact_field_label: The name of the contact field of interest. Rapid_pro uses this to create the
+                                       contact field if it does not exist.
+    :type: target_contact_field_label: str
+    :param rapid_pro: Rapid Pro client to check the contact fields from.
+    :type rapid_pro: rapid_pro_tools.rapid_pro.RapidProClient
+    :param rapid_pro: The target contact field key
+    :type target_contact_field_key: str
+    """
+    workspace_contact_field_labels = [cf.label for cf in workspace_contact_fields]
```

Comment:
The constraint is if the field does not exist we need to create it, and,  rapid pro requires a label to do so. It's also more relatable because we specify labels when creating fields manually in rapidpro.